cmake/compiler/host-gcc/target.cmake

```diff
@@ -14,11 +14,9 @@ find_program(CMAKE_GDB          gdb    )
 # -march={pentium,lakemont,...} do not automagically imply -m32, so
 # adding it here.
 
-# There's only one 64bits ARCH (actually: -mx32). Let's exclude it to
-# avoid a confusing game of "who's last on the command line wins".
 # Maybe the -m32/-miamcu FLAGS should all be next to -march= in the
 # longer term?
-if(NOT CONFIG_X86_64)
+if (CONFIG_X86)
   string(PREPEND CMAKE_ASM_FLAGS             "-m32 ")
   string(PREPEND CMAKE_C_FLAGS               "-m32 ")
```

Comment:
No arch besides X86 and native_posix needed -m32?

Reply:
native_posix does, but I hope setting adding the compiler and linker flags directly from the arch cmake file would do it.